doc/sonic-sub-port-intf-hld.md

```diff
@@ -0,0 +1,493 @@
+    }
+},
+```
+A key in the SUB_INTERFACE table is the name of a sub port, which consists of two sections delimited by a "." (symbol dot).
+The section before the dot is the name of the parent physical port or port channel. The section after the dot is the dot1q encapsulation vlan id.
+
+mtu of a sub port interface is inherited from its parent physical port or port channel, and is not configurable in the current design.
+
+admin_status of a sub port interface can be either up or down.
+In the case field "admin_status" is absent in the config_db.json file, a sub port interface is set admin status up by default at its creation.
+
+Example configuration:
+```
+"SUB_INTERFACE": {
+    "Ethernet64.10": {
+        "admin_status" : "up"
+    },
+    "Ethernet64.10|192.168.0.1/21": {},
+    "Ethernet64.10|fc00::/7": {}
+},
+
+"VLAN": {
+    "Vlan10": {
+        "vlanid" : "10"
+    }
+},
+```
+
+### 2.1.2 CONFIG_DB
+```
+SUB_INTERFACE|{{ port_name }}.{{ vlan_id }}
+    "admin_status" : "{{ adminstatus }}"
+
+SUB_INTERFACE|{{ port_name }}.{{ vlan_id }}|{{ ip_prefix }}
+    "NULL" : "NULL"
+```
+
+### 2.1.3 CONFIG_DB schemas
+```
+; Defines for sub port interface configuration attributes
+key             = SUB_INTERFACE|subif_name      ; subif_name is the name of the sub port interface
+
+; subif_name annotations
+subif_name      = port_name "." vlan_id         ; port_name is the name of parent physical port or port channel
+                                                ; vlanid is DIGIT 1-4094
+
+; field         = value
+admin_status    = up / down                     ; admin status of the sub port interface
+```
+
+```
+; Defines for sub port interface configuration attributes
+key             = SUB_INTERFACE|subif_name|IPprefix     ; an instance of this key will be repeated for each IP prefix
+
+IPprefix        = IPv4prefix / IPv6prefix               ; an instance of this key/value pair will be repeated for each IP prefix
+
+IPv4prefix      = dec-octet "." dec-octet "." dec-octet "." dec-octet "/" %d1-32
+dec-octet       = DIGIT                 ; 0-9
+                 / %x31-39 DIGIT        ; 10-99
+                 / "1" 2DIGIT           ; 100-199
+                 / "2" %x30-34 DIGIT    ; 200-249
+                 / "25" %x30-35         ; 250-255
+
+IPv6prefix      =                             6( h16 ":" ) ls32
+                 /                       "::" 5( h16 ":" ) ls32
+                 / [               h16 ] "::" 4( h16 ":" ) ls32
+                 / [ *1( h16 ":" ) h16 ] "::" 3( h16 ":" ) ls32
+                 / [ *2( h16 ":" ) h16 ] "::" 2( h16 ":" ) ls32
+                 / [ *3( h16 ":" ) h16 ] "::"    h16 ":"   ls32
+                 / [ *4( h16 ":" ) h16 ] "::"              ls32
+                 / [ *5( h16 ":" ) h16 ] "::"              h16
+                 / [ *6( h16 ":" ) h16 ] "::"
+h16             = 1*4HEXDIG
+ls32            = ( h16 ":" h16 ) / IPv4address
+```
+
+Example:
+```
+SUB_INTERFACE|Ethernet64.10
```

Comment:
Would it make more sense to rename the table to SUBPORT?

Reply:
support is more generic than sub interface.  subport cab be non l3 mode which is out of the scope of this proposal